docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+environment that declares what should be running, where, and how the
+deployment should be controlled.
+
+``` {.shell}
+$ aws ecs create-environment --yaml-file <<EOF
+Name: "SomeEnvironment/Prod"
+TaskDefinition: "log-daemon:1"
+Cluster: "default"
+InstanceGroup: 
+  Query: "attribute:stack == prod"
+DeploymentType: Daemon
+DeploymentConfiguration:
+  Method: "Replace" # (or "Surge")
+  MinHealthyPercent: "100"
+EOF
+{ "EnvironmentRevision": 1 }
+```
+
+At this point, nothing has actually happened to your cluster. You can
+see this by showing your environment's status:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Created    0         0         0            0        
+```
+
+In order to make this configuration Active, you first have to Deploy the
+revision:
+
+``` {.shell}
+$ aws ecs deploy-environment --environment-name SomeEnvironment/Prod --environment-revision 1
+{ "DeploymentId": "SomeEnvironment/Prod:1@2017090611472301" }
+```
+
+This will mark the revision as active, and start launching tasks to meet
+the deployment configuration's constraints. In the case of a Daemon
+environment, the desired target is to have one copy of the task for
+every instance in the target Instance Group. For the initial deployment,
+MinHealthyPercent is ignored, since we're starting from a situation that
+already breaches MinHealthyPercent
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         0         0            0        
+```
+
+As the deployment progresses:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         0         0            0        
+```
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         3         3            3         
+```
+
+Failure:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Stuck           5         4         4            3 
+```
+
+Success:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active          5         5         5            5         
+```
+
+Updating an environment
```

Comment:
This section needs to be expanded to show what users will see if the underlying cluster state changes while an update deployment is going on. In particular:

- What happens when new instances join the cluster while an environment is deploying?
- What happens when instances are terminated while an environment is deploying?
- What happens when instances don't come up, or can't be placed (e.g. because the target instance can't accommodate more tasks)?